pkg/apis/core/validation/validation.go

```diff
@@ -4353,6 +4353,19 @@ func ValidateService(service *core.Service) field.ErrorList {
 
 	// external traffic fields
 	allErrs = append(allErrs, validateServiceExternalTrafficFieldsValue(service)...)
+
+	// internal traffic policy field
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceInternalTrafficPolicy) {
+		if service.Spec.InternalTrafficPolicy != nil &&
```

Comment:
Can we put this in a function similar to `validateServiceExternalTrafficFieldsValue`?